prusti-viper/src/encoder/mir/procedures/encoder.rs

```diff
@@ -233,6 +249,134 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
+        location: mir::Location,
+        original_lifetimes: &mut BTreeSet<String>,
+        derived_lifetimes: &mut BTreeMap<String, BTreeSet<String>>,
+    ) -> SpannedEncodingResult<()> {
+        let (new_lifetimes, ended_lifetimes, new_derived_lifetimes) =
+            self.update_lifetimes(original_lifetimes, derived_lifetimes, location);
+        self.encode_end_lft(block_builder, location, ended_lifetimes)?;
+        self.encode_new_lft(block_builder, location, new_lifetimes)?;
+        self.encode_lft_assignment(block_builder, location, new_derived_lifetimes)?;
+        Ok(())
+    }
+
+    fn encode_end_lft(
+        &mut self,
+        block_builder: &mut BasicBlockBuilder,
+        location: mir::Location,
+        lifetimes: BTreeSet<String>,
+    ) -> SpannedEncodingResult<()> {
+        for lifetime in lifetimes {
+            block_builder.add_statement(self.set_statement_error(
+                location,
+                ErrorCtxt::LifetimeEncoding,
+                vir_high::Statement::end_lft_no_pos(lifetime),
+            )?);
+        }
+        Ok(())
+    }
+
+    fn encode_new_lft(
+        &mut self,
+        block_builder: &mut BasicBlockBuilder,
+        location: mir::Location,
+        lifetimes: BTreeSet<String>,
+    ) -> SpannedEncodingResult<()> {
+        for lifetime in lifetimes {
+            block_builder.add_statement(self.set_statement_error(
+                location,
+                ErrorCtxt::LifetimeEncoding,
+                vir_high::Statement::new_lft_no_pos(lifetime),
+            )?);
+        }
+        Ok(())
+    }
+
+    fn encode_lft_assignment(
+        &mut self,
+        block_builder: &mut BasicBlockBuilder,
+        location: mir::Location,
+        lifetimes: BTreeMap<String, BTreeSet<String>>,
+    ) -> SpannedEncodingResult<()> {
+        for (k, v) in lifetimes {
+            if v.len() > 2 {
+                unreachable!("lifetime can't depend on  more than 2 others (yet?)");
+            }
+            let encoded_target = vir_high::VariableDecl::new(k, vir_high::ty::Type::Lifetime {});
+            let encoded_value: vir_high::Expression = if v.len() == 1 {
+                self.enocde_lft_assignment_single(v.iter().next().unwrap().clone())
+            } else {
+                self.enocde_lft_assignment_union(v)
+            };
+            block_builder.add_statement(self.set_statement_error(
+                location,
+                ErrorCtxt::LifetimeEncoding,
+                vir_high::Statement::ghost_assignment_no_pos(encoded_target, encoded_value),
+            )?);
+        }
+        Ok(())
+    }
+
+    fn enocde_lft_assignment_single(&mut self, value: String) -> vir_high::Expression {
+        vir_high::Expression::local_no_pos(vir_high::VariableDecl::new(
+            value,
+            vir_high::ty::Type::Lifetime {},
+        ))
+    }
+
+    fn enocde_lft_assignment_union(&mut self, _values: BTreeSet<String>) -> vir_high::Expression {
```

Comment:
Typo here and in several other method names:
```suggestion
    fn encode_lft_assignment_union(&mut self, _values: BTreeSet<String>) -> vir_high::Expression {
```

Reply:
good catch, done